R/assert_package.R

```diff
@@ -4,70 +4,54 @@
-  if (!is_character_scalar(path)) {
-    return("Invalid package file path")
+assert_package <- function(name, url) {
+  if (!is_character_scalar(name)) {
+    return("Invalid package name.")
   }
-  if (!file.exists(path)) {
-    return(paste("file", shQuote(path), "does not exist"))
-  }
-  name <- trimws(basename(path))
-  url <- try(readLines(path, warn = FALSE), silent = TRUE)
-  if (inherits(url, "try-error")) {
-    return(paste("Problem reading file", shQuote(path)))
+  if (!is_character_scalar(url)) {
+    return("Invalid package URL.")
   }
-  assert_package_contents(name = name, url = url)
-}
-
-assert_package_contents <- function(name, url) {
-  good_package_name <- grepl(
+  name <- trimws(name)
+  url <- trimws(trim_trailing_slash(url))
+  valid_package_name <- grepl(
     pattern = "^[a-zA-Z][a-zA-Z0-9.]*[a-zA-Z0-9]$",
     x = name
   )
-  if (!isTRUE(good_package_name)) {
-    return(paste("Found invalid package name: ", shQuote(name)))
+  if (!isTRUE(valid_package_name)) {
+    return(paste("Found invalid package name:", shQuote(name)))
   }
-  if (!is_character_scalar(url)) {
-    return("Invalid package URL")
-  }
-  url <- trimws(url)
-  good_url <- grepl(
-    pattern = "^https?://[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,3}(/\\S*)?$",
-    x = url
-  )
-  if (!isTRUE(good_url)) {
+  parsed_url <- try(url_parse(url), silent = TRUE)
+  if (inherits(parsed_url, "try-error")) {
     return(
       paste("Found malformed URL", shQuote(url), "of package", shQuote(name))
     )
   }
-}
-
-#' @title Validate a Package URL
-#' @export
-#' @keywords internal
-#' @description Validate that the package URL is in the description file if on
-#'   CRAN.
-#' @return A character string if there is a problem with the URL for the given
-#'   package name, otherwise `NULL` if there are no issues.
-assert_package_url <- function(name, url) {
-  
-  res <- ps(name, size = 1L)
-  pkg <- res[["package"]]
-  if (length(pkg) && name == pkg) {
-    purl <- parse_url(sub("/$", "", url, perl = TRUE))
-    urls <- strsplit(res[["url"]], ",\n|, |\n", perl = TRUE)[[1L]]
-    for (u in urls) {
-      pu <- parse_url(sub("/$", "", u, perl = TRUE))
-      purl[["host"]] == pu[["host"]] && purl[["path"]] == pu[["path"]] &&
-        return(invisible())
-    }
-    burl <- parse_url(res[["bugreports"]])
-    purl[["host"]] == burl[["host"]] &&
-      purl[["path"]] == sub("/issues/*$", "", burl[["path"]], perl = TRUE) &&
-      return(invisible())
+  if (!identical(name, basename(parsed_url[["path"]]))) {
     return(
-      paste("CRAN package", shQuote(name), "does not have URL", shQuote(url))
+      paste(
+        "Package name",
+        shQuote(name),
+        "appears to disagree with the repository name in the URL",
+        shQuote(url)
+      )
     )
   }
-  
+  scheme <- tolower(parsed_url[["scheme"]])
+  if (!identical(scheme, "https")) {
+    return(paste("Scheme of URL", shQuote(url), "is not https."))
+  }
+  host <- tolower(parsed_url[["hostname"]])
```

Comment:
`tolower()` not required on hostname

Reply:
Removed in https://github.com/wlandau/r.releases.utils/commit/4ef45794d266cd6f8f19bb85a632f62090792eea